build/build.go

```diff
@@ -596,20 +597,28 @@ func (s *Session) BuildPackage(pkg *PackageData) (*compiler.Archive, error) {
 		return archive, nil
 	}
 
+	pkgHash := sha256.New()
+
 	if pkg.PkgObj != "" {
-		var fileInfo os.FileInfo
-		gopherjsBinary, err := os.Executable()
-		if err == nil {
-			fileInfo, err = os.Stat(gopherjsBinary)
-			if err == nil {
-				pkg.SrcModTime = fileInfo.ModTime()
-			}
+		binPath, err := os.Executable()
+		if err != nil {
+			return nil, fmt.Errorf("could not locate GopherJS binary: %v", err)
 		}
+		binFile, err := os.Open(binPath)
 		if err != nil {
-			os.Stderr.WriteString("Could not get GopherJS binary's modification timestamp. Please report issue.\n")
-			pkg.SrcModTime = time.Now()
+			return nil, fmt.Errorf("could not open %v: %v", binPath, err)
 		}
 
+		binHash := sha256.New()
+		io.Copy(binHash, binFile)
+		binFile.Close()
```

Comment:
Actually `(*File).Close` doesn't usually cause error (if error happens, this is a critical error), but checking error is not bad here?

Reply:
I'd agree; if that fails we're in serious trouble. I think I'll leave this.